go/clustermetadata/topo/cell_location.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2025 The Multigres Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package topo
+
+import (
+	"context"
+	"errors"
+	"path"
+
+	clustermetadatapb "github.com/multigres/multigres/go/pb/clustermetadata"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// This file provides the utility methods to save / retrieve CellLocation
+// in the topology server.
+//
+// CellLocation records are not meant to be changed while the system is
+// running.  In a running system, a CellLocation can be added, and
+// topology server implementations should be able to read them to
+// access the cells upon demand. Topology server implementations can
+// also read the available CellLocation at startup to build a list of
+// available cells, if necessary. A CellLocation can only be removed if no
+// Shard record references the corresponding cell in its Cells list.
+
+func pathForCellLocation(cell string) string {
+	return path.Join(CellsPath, cell, CellLocationFile)
```

Comment:
Instead of CellLocation, I think we can call it just Cell i.e. CellFile instead of CellLocationFile. It can to begin with only store the cell location, but it might store something more too in which case we're better off having a more generic name